### PR TITLE
feat: add routing of messages between the property inspector and the plugin

### PR DESCRIPTION
**Pull request by GeekyEggo**
Adds routing of messages between the plugin and its property inspector.

### Plugin
```ts
import streamDeck from "@elgato/streamdeck";
streamDeck.ui.registerRoute("/get-names", (req, res) => {
    // access to the request and a responder -alternatively, a response can be returned.
    return ["Mario", "Peach", "Luigi"];
});
```
### Property Inspector Side
```ts
const res = await streamDeck.plugin.fetch("/get-names");
// res.body = ["Mario", "Luigi", "Peach"]
```

### Plugin
- Add `PropertyInspector` class.
  - Add `fetch` to fetch requests from the property inspector.
- Add `streamDeck.ui.current` to track the current property inspector.
- Add `streamDeck.ui.registerRoute` to register the routing of requests _from_ the property inspector.
- Add `@route` decorator to provide routing of an action's function.
- Mark `sendToPlugin` and `sendToPropertyInspector` as deprecated.
### Property Inspector
- Add `streamDeck.plugin fetch` to send fetch requests to the plugin.
- Add `streamDeck.plugin.registerRoute` to register the routing of requests _from_ the plugin.
- Mark `sendToPlugin` and `sendToPropertyInspector` as deprecated.
### Internal
- Add `MessageGateway` class for handling bi-directional requests / responses.
- Update UI namespace to be a controller class.
- Fix trace logging of the connection.

### CODE REVIEW

File: LICENSE
Comment (ChekTek) [inline LICENSE:1]: Is 2024 required here?

File: src/common/messaging/gateway.ts [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+	}
+
+	/**
+	 * Maps the specified {@link path} to the {@link handler}, allowing for requests from the client.
+	 * @param path Path used to identify the route.
+	 * @param handler Handler to be invoked when the request is received.
+	 * @param options Optional routing configuration.
+	 * @returns This instance with the route registered.
+	 */
+	public route<TBody extends JsonValue = JsonValue>(path: string, handler: UnscopedMessageHandler<TAction, TBody>, options?: RouteConfiguration<TAction>): this {
```
Comment (Elgato-AStory) [inline src/common/messaging/gateway.ts:113]: Is there ever any circumstance where we might want to allow a route to be unregistered?

**Review by ChekTek — DISMISSED** [review]

File: LICENSE
Comment (GeekyEggo) [reply LICENSE:1]: Good spot, re-added.

File: src/common/messaging/gateway.ts (hunk 1 above)
Comment (GeekyEggo) [reply src/common/messaging/gateway.ts:113]: There is, but not at the moment. When transient actions are available, and they're added / removed from memory as needed, we'll need to ensure the routes for those context/action combinations are also removed appropriately. This also drove the decision to _not_ return `this` on the publicly exposed `registerRoute` as that will likely return a `Disposable` to allow for the route to be removed.
https://github.com/elgatosf/streamdeck/blob/d41df3a2e75f4dd0dd6843517a076dbd38547a2d/src/plugin/ui/controller.ts#L77-L83

Comment (GeekyEggo) [reply src/common/messaging/gateway.ts]: `registerRoute` now returns an `IDisposable` 🥳

**Review by ChekTek — APPROVED** [review]

**Pull request merged**